components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+            if (validateRegexInput(errorUrlRegex, this.errorUrl)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Setting the Error URL: " + neutralize(this.errorUrl));
+                }
+                // Valid Error URL
+                return true;
+            }
+            // Invalid Error URL
+            return false;
+        }
+        // Error URL not provided
+        return true;
+    }
+
+    /**
+     * Handles the response coming from the framework and decides on whether the response is an authenticated
+     * response or the logout response
+     *
+     * @param messageContext  The message context
+     * @param identityRequest The identity request
+     * @param respBuilder     The response builder
+     * @return
+     */
+    private JWTInboundResponse.JWTInboundResponseBuilder handleFrameworkResponse(
+            IdentityMessageContext messageContext, IdentityRequest identityRequest,
+            JWTInboundResponse.JWTInboundResponseBuilder respBuilder) {
+
+        AuthenticationResult authenticationResult = processResponseFromFrameworkLogin(messageContext,
+                identityRequest);
+        if (log.isDebugEnabled()) {
+            log.debug("Session ID exists.");
+        }
+        if (authenticationResult != null && authenticationResult.isAuthenticated()) {
+            // Authenticated session - response coming from the framework after authentication.
+            respBuilder = handleAuthenticationResult(identityRequest, authenticationResult, respBuilder);
+        } else {
+            // Non-authenticated session - response coming after logged out - redirecting the user to Logout URL
+            respBuilder = handleLogoutResult(identityRequest, respBuilder);
+        }
+        return respBuilder;
+    }
+
+    /**
+     * Handles the response coming from the framework after authentication.
+     *
+     * @param identityRequest      The identity request
+     * @param authenticationResult The authentication result
+     * @param respBuilder          The response builder
+     * @return The response builder after setting the required options
+     */
+    private JWTInboundResponse.JWTInboundResponseBuilder handleAuthenticationResult(
+            IdentityRequest identityRequest, AuthenticationResult authenticationResult,
+            JWTInboundResponse.JWTInboundResponseBuilder respBuilder) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Authenticated session exists. Treating as a request coming after the " +
+                    "authentication.");
+        }
+        String userName = authenticationResult.getSubject().getUserName();
+        if (log.isDebugEnabled()) {
+            log.debug("Processing the request for the user: " + neutralize(userName));
+        }
+
+        // Validate API key and Generate JWT Token
+        Map<ClaimMapping, String> userAttributes = authenticationResult.getSubject().getUserAttributes();
```

Comment:
Ideally, this validation should be done before the request is sent to the framework for handling user authentication